test/functional/test_framework/test_base.py

```diff
@@ -0,0 +1,80 @@
+"""Tests for the base metaclass
+
+The tests contained within this module ensure that the base metaclass for
+tests will raise a ``TypeError`` if either:
+
+    - Subclasses try to override either `__init__` or `main` OR
+    - Subclasses do not override the `run_test` and `set_test_params` methods
+
+The metaclass should allow classes adhering to the above standards to be
+created.
+"""
+import unittest
+
+from base import BitcoinTestMetaClass
+
+
+class TestMetaClass(unittest.TestCase):
+
+    def test_allows_valid_class(self):
+        class Foo(metaclass=BitcoinTestMetaClass):
+            def run_test(self):
+                pass
+
+            def set_test_params(self):
+                pass
+
+    def test_raises_for_init(self):
+        msg = ("BitcoinTestFramework subclasses may not override "
+               "'__init__' or 'main'")
+
+        with self.assertRaises(TypeError, msg=msg):
+            class Foo(metaclass=BitcoinTestMetaClass):
+                def __init__(self):
+                    pass
+
+                def run_test(self):
+                    pass
+
+                def set_test_params(self):
+                    pass
+
+    def test_raises_for_main(self):
+        msg = ("BitcoinTestFramework subclasses may not override "
+               "'__init__' or 'main'")
+
+        with self.assertRaises(TypeError, msg=msg):
+            class Foo(metaclass=BitcoinTestMetaClass):
+                def main(self):
+                    pass
+
+                def run_test(self):
+                    pass
+
+                def set_test_params(self):
+                    pass
+
+    def test_raises_if_no_run_test(self):
+        msg = ("BitcoinTestFramework subclasses must override "
+               "'run_test' and 'set_test_params'")
+
+        with self.assertRaises(TypeError, msg=msg):
+            class Foo(metaclass=BitcoinTestMetaClass):
+                def set_test_params(self):
+                    pass
+
+    def test_raises_if_no_set_test_params(self):
+        msg = ("BitcoinTestFramework subclasses must override "
+               "'run_test' and 'set_test_params'")
+
+        with self.assertRaises(TypeError, msg=msg):
+            class Foo(metaclass=BitcoinTestMetaClass):
+                def run_test(self):
+                    pass
+
+    def test_exempts_framework_class(self):
+        class BitcoinTestFramework(metaclass=BitcoinTestMetaClass):
+            pass
+
+if __name__ == '__main__':
+    unittest.main()
```

Comment:
As is this is not integrated into your current CI. Still trying to wrap my head around everything so I could be wrong on this but since you are using your own test suite / runner which uses this meta class I don't think there's a great way to integrate into your current workflow. This can be tested directly from the command line as needed